### PR TITLE
COMMERCE-6852 CPDefinition resource permission in channelRel should be checked in the service

### PR DESCRIPTION
**Pull request by riccardo-alberti**

### CODE REVIEW

**Comment by riccardo-alberti** [comment]
ci:test:sf

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Review by dsitu — APPROVED** [review]

**Comment by riccardo-alberti** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CommerceChannelRelServiceImpl.java [hunk 1]
```diff
@@ -125,16 +126,26 @@ public CommerceChannelRel getCommerceChannelRel(long commerceChannelRelId)
...
 
 	@Override
 	public List<CommerceChannelRel> getCommerceChannelRels(
-		String className, long classPK, String name, int start, int end) {
+			String className, long classPK, String name, int start, int end)
+		throws PortalException {
+
+		if (className.equals(CPDefinition.class.getName())) {
+			_cpDefinitionModelResourcePermission.check(
+				getPermissionChecker(), classPK, ActionKeys.UPDATE);
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CommerceChannelRelServiceImpl.java:144]: Why UPDATE and not VIEW permission?

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CommerceChannelRelServiceImpl.java [hunk 2]
```diff
@@ -164,14 +175,25 @@ public int getCommerceChannelRelsCount(long commerceChannelId)
...
 
 	@Override
 	public int getCommerceChannelRelsCount(
-		String className, long classPK, String name) {
+			String className, long classPK, String name)
+		throws PortalException {
+
+		if (className.equals(CPDefinition.class.getName())) {
+			_cpDefinitionModelResourcePermission.check(
+				getPermissionChecker(), classPK, ActionKeys.UPDATE);
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CommerceChannelRelServiceImpl.java:192]: Same question here

**Pull request closed without merging**